src/sentry/services/hybrid_cloud/app/__init__.py

```diff
@@ -30,35 +32,32 @@ class RpcSentryAppService:
     service_type: str = "sentry_app"
 
 
-@dataclass
-class RpcSentryApp:
+class RpcSentryApp(RpcModel):
     id: int = -1
-    scope_list: List[str] = field(default_factory=list)
+    scope_list: List[str] = Field(default_factory=list)
     application_id: int = -1
     proxy_user_id: Optional[int] = None  # can be null on deletion.
     owner_id: int = -1  # relation to an organization
     name: str = ""
     slug: str = ""
     uuid: str = ""
-    events: List[str] = field(default_factory=list)
+    events: List[str] = Field(default_factory=list)
 
 
-@dataclass
-class RpcSentryAppInstallation:
+class RpcSentryAppInstallation(RpcModel):
     id: int = -1
     organization_id: int = -1
     status: int = SentryAppInstallationStatus.PENDING
-    sentry_app: RpcSentryApp = field(default_factory=lambda: RpcSentryApp())
+    sentry_app: RpcSentryApp = Field(default_factory=lambda: RpcSentryApp())
     date_deleted: Optional[datetime.datetime] = None
     uuid: str = ""
 
 
-@dataclass
-class RpcSentryAppComponent:
+class RpcSentryAppComponent(RpcModel):
     uuid: str = ""
     sentry_app_id: int = -1
     type: str = ""
-    schema: Mapping[str, Any] = field(default_factory=dict)
+    app_schema: Mapping[str, Any] = Field(default_factory=dict)
```

Comment:
We need a name change here because `schema` is inherited as a `pydantic.BaseModel` attribute.

Reply:
Unfortunately no, it's not only a problem with the dynamic model classes. The `RpcSentryAppComponent` class, right here and now, is extending `RpcModel`, which extends `pydantic.BaseModel`. Declaring a `schema` attribute overrides `BaseModel`'s built-in one, and I believe there's a metaclass which detects this and complains.

Personally, I don't think this is worth any special gymnastics. By extending `RpcModel` rather than making simple dataclasses, we inherit methods named
```
"construct", "copy", "dict", "from_orm", "json", "parse_file", "parse_obj",
"parse_raw", "schema", "schema_json", "update_forward_refs", "validate"
```
Several of those, we're already calling elsewhere. We won't have any trouble as long as we treat those as reserved names (and Pydantic will fail early if we forget).

Reply:
Okie doke